ambry-protocol/src/test/java/com/github/ambry/protocol/PutRequestTest.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.clustermap.MockPartitionId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.messageformat.BlobType;
+import com.github.ambry.utils.NettyByteBufDataInputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class PutRequestTest {
```

Comment:
nit: javadocs?
Also this is not PutRequestTest. This PutRequest v5 deserialization test.

Reply:
I'll add the Java doc.  This class is for PutRequest testing.  It starts with only 1 unit test instead of testing the entire class.  In the future, anyone modifying the PutRequest class should add their unit test here.  The unit test inside this class make sure it can serialize a put request and de-serialize a put request.  Serialization always use the latest format version, so the de-serialized output should match the latest version.  Again, this unit test is not intended to test all versions.  That's the job of other unit tests that test each version.  This unit test should focus on testing PutRequest only.  BTW, we could add more unit tests as needed.